build/templates/session.py.mako

```diff
@@ -286,9 +226,7 @@ class Session(_SessionBase):
         # Initialize the superclass with default values first, populate them later
         super(Session, self).__init__(
             repeated_capability_list=[],
-            ${config['session_handle_parameter_name']}=None,
-            library=None,
-            encoding=None,
+            library_interpreter=None,
```

Comment:
Just following what was done before. I might change it anyway, though. I'll leave this conversation open for now.

Reply:
> Just following what was done before.

Yeah, it was always ugly.

How do you think we should change it? One half-baked idea is to add smarts to codegenerator so that LibraryInterpreter methods know to store the session handle returned from Init functions automatically. This way Session can become almost completely unaware of session handles. What do you think?

Reply:
> How do you think we should change it?

For this particular conversation ("Can't LibraryInterpreter be constructed as we invoke `super.__init__`?"), it's going to just be "move the object construction up a few lines, to happen while we're initializing SessionBase."

But that doesn't cover the session handle ugliness, which is a different issue. For that, I agree that doing some special Init magic at the LibraryInterpreter layer might work. A simpler solution for solving the encapsulation issue, though, is adding a set_session_handle method to the LibraryInterpreter class. I don't mind just leaving it as-is, though. LibraryInterpreter is an internal class, and if we find some reason in the future to fix the encapsulation, it'll be easy enough to change at that time.

Reply:
Moved the LibraryInterpreter construction.

Let me know what you think about the session handle stuff. I think that's all that's left, other than the hapigen updates.

Reply:
We can leave it alone for this PR, and revisit if we want/need to.